braindecode/datautil/transforms.py

```diff
@@ -10,9 +10,76 @@
 #
 # License: BSD (3-clause)
 
+from collections import OrderedDict
 import mne
 
 
+def transform_concat_ds(concat_ds, transforms):
+    """Apply a number of transformers to a concat dataset.
+
+    Parameters
+    ----------
+    concat_ds: A concat of BaseDataset or WindowsDataset
+        datasets to be transformed
+    transforms: dict(str | callable: dict)
+        dict with function names of mne.raw or a custom transform and function
+        kwargs
+
+    Returns
+    -------
+    concat_ds:
+    """
+    if not isinstance(transforms, OrderedDict):
+        raise TypeError(
+            "Order of transforms matters! Please provide an OrderedDict.")
+    for ds in concat_ds.datasets:
+        if hasattr(ds, "raw"):
+            _transform_raw(ds.raw, transforms)
+        elif hasattr(ds, "windows"):
+            _transform_windows(ds.windows, transforms)
+        else:
+            raise ValueError(
+                'Can only transform concatenation of BaseDataset or '
+                'WindowsDataset, with either a `raw` or `windows` attribute.')
+
+    # Recompute cumulative sizes as the transforms might have changed them
+    # XXX: Ultimately, the best solution would be to have cumulative_size be
+    #      a property of BaseConcatDataset.
+    concat_ds.cumulative_sizes = concat_ds.cumsum(concat_ds.datasets)
+
+
+def _transform_raw(raw, transforms):
+    for transform, transform_kwargs in transforms.items():
+        if callable(transform):
+            _custom_transform_raw(raw, transform, transform_kwargs)
+        else:
+            if not hasattr(raw, transform):
+                raise AttributeError(f"Raw does not have {transform}.")
+            _mne_transform(raw, transform, transform_kwargs)
+
+
+def _custom_transform_raw(raw, transform, transform_kwargs):
+    transform(raw, **transform_kwargs)
+
+
+def _mne_transform(raw_or_epochs, transform, transform_kwargs):
+    getattr(raw_or_epochs.load_data(), transform)(**transform_kwargs)
+
+
+def _transform_windows(windows, transforms):
+    for transform, transform_kwargs in transforms.items():
+        if callable(transform):
+            _custom_transform_windows(windows, transform, transform_kwargs)
+        else:
+            if not hasattr(windows, transform):
+                raise AttributeError(f"Epochs does not have {transform}.")
+            _mne_transform(windows, transform, transform_kwargs)
+
+
+def _custom_transform_windows(windows, transform, transform_kwargs):
+    transform(windows, **transform_kwargs)
```

Comment:
`_transform_raw` and `_transform_windows` as well as `_custom_transform_raw` and `_custom_transform_windows` could be merged into one function I think. The differentiation at line 35 is just used for the error messages and could be done inside the merged function.

Reply:
Good point, I'll merge them!